tests/pipeline/controlflowanalysis/test_variable_name_generation.py

```diff
@@ -0,0 +1,121 @@
+    for index, var_type in enumerate(ALL_TYPES):
+        assignments.append(Assignment(Variable(f"var_{index}", var_type), Constant(0)))
+        assignments.append(Assignment(Variable(f"var_p_{index}", Pointer(var_type)), Constant(0)))
+    return assignments
+
+
+def _generate_options(notation: str = "system_hungarian", pointer_base: bool = True, type_sep: str = "", counter_sep: str = "") -> Options:
+    options = Options()
+    options.set(f"{PIPELINE_NAME}.notation", notation)
+    options.set(f"{PIPELINE_NAME}.pointer_base", pointer_base)
+    options.set(f"{PIPELINE_NAME}.type_separator", type_sep)
+    options.set(f"{PIPELINE_NAME}.counter_separator", counter_sep)
+    options.set(f"code-generator.max_complexity", 100)
+    options.set("code-generator.use_increment_int", False)
+    options.set("code-generator.use_increment_float", False)
+    options.set("code-generator.use_compound_assignment", True)
+    return options
+
+
+def _run_vng(ast: AbstractSyntaxTree, options: Options = _generate_options()):
+    task = DecompilerTask("variable_name_generation", None, ast, options, VOID)
+    VariableNameGeneration().run(task)
+    DecoratedCode.print_code(CodeGenerator().generate([task]))
+
+
+def test_default_notation_1():
+    true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
+    ast = AbstractSyntaxTree(CodeNode(Assignment(var := Variable("var_0", I32), Constant(0)), true_value), {})
+    _run_vng(ast, _generate_options(notation="default"))
+    assert var.name == "var_0"
+
+
+class TestHungarianNotation:
+    def test_hungarian_notation_all_types(self):
+        true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
+        ast = AbstractSyntaxTree(cn := CodeNode(_generate_all_type_assignments(), true_value), {})
+        _run_vng(ast)
+        assert [str(_) for _ in cn.instructions] == [
+            "chVar0 = 0x0",
+            "chpVar0 = 0x0",
+            "sVar1 = 0x0",
+            "spVar1 = 0x0",
+            "iVar2 = 0x0",
+            "ipVar2 = 0x0",
+            "lVar3 = 0x0",
+            "lpVar3 = 0x0",
+            "i128Var4 = 0x0",
+            "i128pVar4 = 0x0",
+            "uchVar5 = 0x0",
+            "uchpVar5 = 0x0",
+            "usVar6 = 0x0",
+            "uspVar6 = 0x0",
+            "uiVar7 = 0x0",
+            "uipVar7 = 0x0",
+            "ulVar8 = 0x0",
+            "ulpVar8 = 0x0",
+            "ui128Var9 = 0x0",
+            "ui128pVar9 = 0x0",
+            "hVar10 = 0x0",
+            "hpVar10 = 0x0",
+            "fVar11 = 0x0",
+            "fpVar11 = 0x0",
+            "dVar12 = 0x0",
+            "dpVar12 = 0x0",
+            "ldVar13 = 0x0",
+            "ldpVar13 = 0x0",
+            "qVar14 = 0x0",
+            "qpVar14 = 0x0",
+            "oVar15 = 0x0",
+            "opVar15 = 0x0",
+            "bVar16 = 0x0",
+            "bpVar16 = 0x0",
+            "vVar17 = 0x0",
+            "vpVar17 = 0x0",
+        ]
```

Comment:
Kinda creative way to test this.
Would it be possible to use [parameterize](https://docs.pytest.org/en/6.2.x/parametrize.html) instead to have clear links between the input and the expected output?

e.g.:
```python
import pytest

@pytest.mark.parametrize(
    "variable,name",
    [
        (Variable(x, Integer.char()), "cVar0"),
        ...
    ],
)
def test_hungarian_notation(variable, name):
```

at the moment, this seems more like an acceptence test rather than unittests. If you want to be more generic with regards to your options, you should test parts of the Renaming on their own e.g. test_hungarian_prefix, test_hungarian_suffix etc.

Reply:
First part refactored. Every combination will be checked. 
Will add a few more tests for the second part later.